### PR TITLE
ESQL: Use Point geometry in tests

### PR DESCRIPTION
**Pull request by iverase**
We are currently use the SpatialPoint abstraction on ES|QL geo test which makes it difficult to integrate with geo_shape. Therefore this PR changes the implementation to use Point geometries.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 1]
```diff
@@ -1194,34 +1192,6 @@ public static String randomDateFormatterPattern() {
         return randomFrom(FormatNames.values()).getName();
     }
 
-    /**
-     * Generate a random valid point constrained to geographic ranges (lat, lon ranges).
-     */
-    public static SpatialPoint randomGeoPoint() {
```
Comment (craigtaverner) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:1200]: I had intended to use these to replace a number of cases in Elasticsearch test code where we generate random GeoPoint (and cartesian Point), but was going to do that as a separate PR later. However, I can probably do that anyway, at low priority.

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java [hunk 2]
```diff
@@ -162,31 +161,7 @@ protected void assertResults(
...
-                sb.append(valueToString(field));
-            }
-            return sb.append("]").toString();
-        } else if (value instanceof SpatialPoint point) {
-            // Alternatively we could just change GeoPoint.toString() to use WKT, but that has other side-effects
-            return point.toWKT();
-        } else {
-            return value.toString();
-        }
+        assertData(expected, actualValues, testCase.ignoreOrder, logger, value -> value == null ? "null" : value.toString());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java:164]: This is much nicer!

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java [hunk 3]
```diff
@@ -36,85 +34,23 @@ public long pointAsLong(double x, double y) {
         }
     },
     CARTESIAN {
-        public SpatialPoint longAsPoint(long encoded) {
-            try {
-                final double x = XYEncodingUtils.decode((int) (encoded >>> 32));
-                final double y = XYEncodingUtils.decode((int) (encoded & 0xFFFFFFFF));
-                return makePoint(x, y);
-            } catch (Error e) {
-                throw new IllegalArgumentException("Failed to convert invalid encoded value to cartesian point");
```
Comment (craigtaverner) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java:45]: I seem to remember that this exception wrapping was needed for some correct ESQL conversion of errors to null plus warnings. We needed explicitly to use IllegalArgumentException for that. Hopefully there are tests that will fail if this is not correctly done.

**Review by craigtaverner — APPROVED** [review]
Generally I think the move to Point is a great idea, and a big cleanup. There are a few bits I worry about, particularly the removal of exception wrapping to IllegalArgumentException, which I think was needed for ESQL conversion of exceptions to null results with warnings. But we can also verify that with tests. I think the `ToCartesianPointTests` should cover this case.

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java (hunk 1 above)
Comment (iverase) [reply test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:1200]: I don't think we should have geo specific code in this class.

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java (hunk 3 above)
Comment (iverase) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java:45]: If there is no test, there is no issue :)

**Pull request merged**